src/jukebox/components/synchronisation/sync_shared/__init__.py

```diff
@@ -0,0 +1,355 @@
+                return
+
+            with cfg_sync_shared:
+                self._sync_on_rfid_scan_enabled = (cfg_sync_shared.getn('sync_shared', 'on_rfid_scan_enabled', default=False)
+                                                is True)
+                if not self._sync_on_rfid_scan_enabled:
+                    logger.info("Sync on RFID scan deactivated")
+                self._sync_mode = cfg_sync_shared.getn('sync_shared', 'mode')
+                self._sync_remote_server = cfg_sync_shared.getn('sync_shared', self._sync_mode, 'server')
+                self._sync_remote_port = int(cfg_sync_shared.getn('sync_shared', self._sync_mode, 'port'))
+                self._sync_remote_timeout = int(cfg_sync_shared.getn('sync_shared', self._sync_mode, 'timeout'))
+                self._sync_remote_path = cfg_sync_shared.getn('sync_shared', self._sync_mode, 'path')
+
+                self._sync_is_mode_ssh = self._sync_mode == "ssh"
+                if self._sync_is_mode_ssh:
+                    self._sync_remote_ssh_user = cfg_sync_shared.getn('sync_shared', self._sync_mode, 'username')
+
+            components.rfid.reader.rfid_card_detect_callbacks.register(self.rfid_callback)
+        else:
+            logger.info("Sync shared deactivated")
+
+    def __exit__(self):
+        cfg_sync_shared.save(only_if_changed=True)
+
+    def rfid_callback(self, card_id: str, state: int):
+        if state == -1:
+            self.sync_card_database(card_id)
+
+    @plugs.tag
+    def sync_change_on_rfid_scan(self, option: str = 'toggle') -> None:
+        """
+        Change activation of 'on_rfid_scan_enabled'
+
+        :param option: Must be one of 'true', 'false', 'toggle'
+        """
+        if self._sync_enabled:
+
+            if option == 'true':
+                _new_state = True
+            elif option == 'false':
+                _new_state = False
+            elif option == 'toggle':
+                _new_state = not self._sync_on_rfid_scan_enabled
+            else:
+                logger.error(f"Invalid value '{option}' for 'option' in 'sync_change_on_rfid_scan'")
+                _new_state = None
+
+            if _new_state is not None:
+                cfg_sync_shared.setn('sync_shared', 'on_rfid_scan_enabled', value=_new_state)
+                self._sync_on_rfid_scan_enabled = _new_state
+
+                logger.info(f"Changed 'on_rfid_scan_enabled' to '{_new_state}'")
+
+        else:
+            logger.debug("Sync shared deactivated")
+
+    @plugs.tag
+    def sync_full(self) -> bool:
+        """
+        Sync full from the remote server
+        """
+        _files_synced = False
+
+        if self._sync_enabled:
+            logger.info("Syncing full")
+            _database_synced = self._sync_card_database()
+            _folder_synced = self._sync_folder('')
+            _files_synced = _database_synced or _folder_synced
```

Comment:
I'd expect this to be one single call, maybe with an array of paths (files are also just paths)

```
paths = [ path_to_db, path_to_audiofolder, path_to_something_else ]
for path in paths:
    self.sync(path)
```

Reply:
They are not quite the same. The audiofolders are just placed in the file system and mpd can performe the update after.
The card database behaves a bit different as its a singel file for all cards, and therefore a special treatment is needed to update correctly (and only if changes are present). See also the comment in the header, where i tried to describe this specialities.

Reply:
Just in case of "sync_all", where the card.yaml is just replaced.
But in case of "sync_on_scan" only the entry of this single CardId has to be updated. Otherwise it would break the offline functionality, as other cards could have been changed, but the audiofiles are not yet present on the phoniebox.